src/Agent.Plugins/PipelineCache/PipelineCacheServer.cs

```diff
@@ -150,18 +282,125 @@ public class PipelineCacheServer
             return pipelineCacheClient;
         }
 
-        private Task DownloadPipelineCacheAsync(
+        private async Task DownloadPipelineCacheAsync(
+            AgentTaskPluginExecutionContext context,
             DedupManifestArtifactClient dedupManifestClient,
             DedupIdentifier manifestId,
             string targetDirectory,
+            bool isTar,
+            string contentFormat,
             CancellationToken cancellationToken)
         {
-            DownloadDedupManifestArtifactOptions options = DownloadDedupManifestArtifactOptions.CreateWithManifestId(
-                manifestId,
-                targetDirectory,
-                proxyUri: null,
-                minimatchPatterns: null);
-            return dedupManifestClient.DownloadAsync(options, cancellationToken);
+
+            // Throw for bunch of invalid combinations of istar and contentFormat ??
+            if (!isTar)
+            {
+                DownloadDedupManifestArtifactOptions options = DownloadDedupManifestArtifactOptions.CreateWithManifestId(
+                    manifestId,
+                    targetDirectory,
+                    proxyUri: null,
+                    minimatchPatterns: null);
+
+                await dedupManifestClient.DownloadAsync(options, cancellationToken);
+            }
+            else
+            {
+                string manifestPath = Path.Combine(Path.GetTempPath(), $"{nameof(DedupManifestArtifactClient)}.{Path.GetRandomFileName()}.manifest");
+                using (var manifestStream = File.Create(manifestPath))
+                {
+                    await dedupManifestClient.DownloadToStreamAsync(manifestId, manifestStream, null, cancellationToken);
+                }
```

Comment:
There's a problem with using stream now that I think about it. Use DownloadToFilePath instead and feed that into JsonSerializer instead.

Reply:
Sure. What's the problem with stream btw?

Reply:
I'm worried that the downloader might be stuck on writing to the stream and if the JsonSerializer might terminate prematurely because the stream isn't done writing yet.